test/params_test.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "testutil.h"
+#include "internal/nelem.h"
+#include <openssl/params.h>
+
+static int test_params(void)
+{
+    int8_t i8;
+    size_t sz;
+    double dbl;
+    const OSSL_PARAM params[] = {
+        OSSL_PARAM_int8("a", &i8),
+        OSSL_PARAM_size_t("b", &sz),
+        OSSL_PARAM_double("c", &dbl)
+    };
```

Comment:
May I suggest testing your API against arrays that have been set up in raw form, as well as an param request that's responded to in raw forms?

Reply:
Yes, I like this suggestion I'll do that in addition to other tests.

Reply:
Endian issues come into play with raw forms.  Test cases will need to be endian aware.

Reply:
Yup

Reply:
Added